### ISSUE TITLE
MSVCP140.dll error in Windows 10

### ISSUE THREAD

**Issue by auchenberg** [MarkTiedemann/fastlist#1]
I have a problem with a Windows 10 user using https://github.com/sindresorhus/ps-list#readme on Windows:

<img width="342" alt="Screen Shot 2020-06-02 at 7 43 44 PM" src="https://user-images.githubusercontent.com/173559/83589928-84d40e80-a509-11ea-8688-3ff58e7a0a5d.png">

Is this a known issue? I assume there's some dependencies on a C++ lib/runtime?

**Comment by MarkTiedemann** [comment]
I believe this `.dll` is part of the Universal C Runtime Library (UCRT) which is a system component managed by Windows Update. The C runtime is used in fastlist using `#include <iostream>` to stream output to stdout (`std::cout << ...`).

There are multiple temporary work-around options:
- Using fastlist v0.1.0 (which is included in ps-list v7.0.0, rather than ps-list v7.1.0 which uses fastlist v0.2.1)
- (Re-)Installing the Visual C++ Redist for VS as described here: https://github.com/MarkTiedemann/fastlist#compatibility
- Running the Windows System File Checker (SFC) to check for corrupted system files: `sfc /scannow`

I hope this helps as a first triage. I will investigate more later today. Perhaps the `#include <iostream>; ` has to be removed and replaced with a good old `#include <stdio.h>; printf(...)` to avoid this issue...

**Comment by MarkTiedemann** [comment]
@auchenberg Thanks again for reporting this issue!

I am able to reproduce it using a fresh Windows Docker container.

To make sure that this will not happen again, I have added a [`test.cmd`](https://github.com/MarkTiedemann/fastlist/blob/master/test.cmd) script:

The old fastlist v0.1.0 passes the test:
```batch
:: fastlist v0.1.0
C:\dev\fastlist>test
[System Process]        0       0
System  4       0
:: ...
cmd.exe 1652    1788
fastlist.exe    1620    1652

Info: Test passed
```

However, fastlist v0.2.1 fails the test:
```batch
:: fastlist v0.2.1
C:\dev\fastlist>test
Error: Test failed with error level: -1073741515
```

Using `depends.exe` from https://www.dependencywalker.com/, I was also able to check which `.dll`s fastlist depends on.

v0.1.0 only depends on `kernel32.dll` (which is unavoidable). v0.2.1 depends on a whole bunch of `.dll`s. See screenshot below:

![Capture](https://user-images.githubusercontent.com/7062160/83802488-2777c300-a6ab-11ea-990e-ed7e436be8ef.PNG)

Previously I built fastlist v0.1.0 using the command line tools from [Visual Studio 2010](https://github.com/lukesampson/scoop-extras/blob/master/bucket/visualc.json). For fastlist v0.2.0 I upgraded to VS 2019. Unfortunately, it turns out that the MSVC toolchain from 2010 links statically by default, which is not the case for VS 2019.

I will release a fix in the coming days. In the meantime, I can recommend to use fastlist v0.1.0, which is included in [`ps-list`](https://github.com/sindresorhus/ps-list) v7.0.0, or, alternatively, [`tasklist`](https://github.com/sindresorhus/tasklist), which I maintain together with Sindre and which will also give you a list of running processes on Windows, albeit slower.

I am very sorry for any inconvenience that this may have caused.

**Comment by auchenberg** [comment]
Thank you so much for taking the time to debug and look into this @MarkTiedemann. Your work is much appreciated!

For our project we switched to `tasklist` on Windows to avoid the distribution and execution of the executable for now.

**Comment by MarkTiedemann** [comment]
> Thank you so much for taking the time to debug and look into this @MarkTiedemann. Your work is much appreciated!

*blushes* Thank you, too! Your bug report is a very valuable contribution to this project since this is indeed a major bug.

> For our project we switched to tasklist on Windows to avoid the distribution and execution of the executable for now.

The `tasklist` Node module executes `%System32%\tasklist.exe`. So you only avoid distribution, not execution of an executable.

**Comment by MarkTiedemann** [comment]
Closed in [v0.3.0](https://github.com/MarkTiedemann/fastlist/releases/tag/v0.3.0).

**Comment by MarkTiedemann** [comment]
Just tested fastlist v0.3.0 on my old 32-bit Windows Vista laptop from 2006. Works like a charm. :)
